app/views/topics/new.html.erb

```diff
@@ -0,0 +1,8 @@
+<h2>Topics</h2>
+<p>Add a new topic</p>
+
+<%= form_for @topic, url: {action: "create"}, html: {class: "nifty_form"} do |f| %>
```

Comment:
You don't need that "nifty_form" class yet? [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)

Reply:
For CSS use [BEM](http://getbem.com/) classes
For selecting elements with Javascript don't use classes: use data attributes instead.

Reply:
The latest Rails form helper is `form_with`, so try using that. Here's a write up of how it relates to the old form helpers: https://m.patrikonrails.com/rails-5-1s-form-with-vs-old-form-helpers-3a5f72a8c78a

Reply:
Regarding `action: "create"`, remember to use symbols wherever possible in Ruby - They take up less memory.

For example...
```ruby
variableOne = "my string"
variableTwo = "my string"
variableThree = "my string"
```

```ruby
variableOne = :my_string
variableTwo = :my_string
variableThree = :my_string
```

In the first example three strings are created in memory, whereas, in the second example, only one symbol is created in memory (the three different variables all point to the same symbol)

There is even a denial of service attack based on overloading the servers memory with the creation of millions of strings. Using symbols instead of strings protects against this.